polyfills/Symbol/tests.js

```diff
@@ -0,0 +1,165 @@
+// http://people.mozilla.org/~jorendorff/es6-draft.html#sec-symbol-constructor
+it('should throw if being used via `new`', function() {
+	var test = function () {
+		return new Symbol();
+	}
+	expect(test).to.throwError();
+});
+
+it('should have Symbol as the constructor property on the prototype', function() {
+	expect(Object.getPrototypeOf(Symbol()).constructor).to.be(Symbol)
+});
+
+it('should silently fail when assigning new properties', function (){
+	const a = Symbol('a');
+	a.b = '1';
+	expect(a.b).to.be.undefined;
+});
+
+it('should have Symbol.prototype as the prototype of an instance', function() {
+	expect(Object.getPrototypeOf(Symbol())).to.be(Symbol.prototype);
+});
+
+it('should return "[object Symbol]" when called with Object.prototype.toString()', function() {
+	expect(Object.prototype.toString.call(Symbol())).to.be('[object Symbol]');
+});
+
+it('should silently fail when overwriting properties', function() {
+	var sym = Symbol('66');
+	sym.toString = 0;
+	expect(sym.toString).to.be.a(Function)
+	sym.valueOf = 0;
+	expect(sym.valueOf).to.be.a(Function)
+});
+
+it('should create unique symbols', function() {
+	expect(Symbol('a')).to.not.equal(Symbol('a'))
+});
+
+it('has for, and keyFor static methods', function() {
+	expect(Symbol.for).to.be.a(Function);
+	expect(Symbol.keyFor).to.be.a(Function);
+});
+
+it('Symbol.keyFor should throw if not given a symbol', function() {
+	var stringKeyFor = function() {
+		return Symbol.keyFor('a');
+	}
+	var numberKeyFor = function() {
+		return Symbol.keyFor(1);
+	}
+	var arrayKeyFor = function() {
+		return Symbol.keyFor([]);
+	}
+	var objectKeyFor = function() {
+		return Symbol.keyFor({});
+	}
+	var boolKeyFor = function() {
+		return Symbol.keyFor(true);
+	}
+	var undefinedKeyFor = function() {
+		return Symbol.keyFor(undefined);
+	}
+	var symbolKeyFor = function() {
+		return Symbol.keyFor(Symbol('a'));
+	}
+
+	expect(stringKeyFor).to.throwError();
+	expect(numberKeyFor).to.throwError();
+	expect(arrayKeyFor).to.throwError();
+	expect(objectKeyFor).to.throwError();
+	expect(boolKeyFor).to.throwError();
+	expect(undefinedKeyFor).to.throwError();
+	expect(symbolKeyFor).to.not.throwError();
+});
+
+it('Symbol.keyFor should return undefined if can not find symbol in global registry', function() {
+	expect(Symbol.keyFor(Symbol('a'))).to.be.undefined;
```

Comment:
Isn't there a possibility that 'a' could be in the global symbol registry?  Not sure we can guarantee the order of execution: the test at line 85 running before this test might cause this one to fail erroneously.